src/main/java/games/strategy/engine/data/GameParser.java

```diff
@@ -738,20 +738,21 @@ private void parseResources(final Element root) throws GameParseException {
     for (final Element element : getChildren("resource", root)) {
       final String name = element.getAttribute("name");
       final String isDisplayedFor = element.getAttribute("isDisplayedFor");
-      final List<PlayerID> players = data.getPlayerList().getPlayers();
-      if (isDisplayedFor.equalsIgnoreCase(RESOURCE_IS_DISPLAY_FOR_NONE)) {
-        players.clear();
-      } else if (!isDisplayedFor.isEmpty()) {
-        players.clear();
+      if (isDisplayedFor.isEmpty()) {
+        data.getResourceList().addResource(new Resource(name, data, data.getPlayerList().getPlayers()));
+      } else if (isDisplayedFor.equalsIgnoreCase(RESOURCE_IS_DISPLAY_FOR_NONE)) {
```

Comment:
For name of variable: `RESOURCE_IS_DISPLAY_FOR_NONE` is not bad. A suggestion for maybe a better one:  `DEFAULT_NO_VALUE`

Reply:
Problem with that is this class covers all parsing so it would need to cover default no value or none for all cases. Without checking all of the XML parameters, I'm not sure that is the case.

Reply:
Ah, we may be looking at this from a different perspective. I'm not associating personally the 'NONE' value specifically to this one attribute. From what I can tell `NONE` is a magic valued supplied by the XML parser as we have the attribute defined as 'IMPLIED'. For me the concern is then knowing where 'NONE' came from. I'm pretty sure on a naive pass at this code, not looking at it in the context of the this PR, I would have probably assumed 'NONE' was going to be an explicitly defined value in XML. So really the recommendation is to make the magic 'NONE' value more explicit in one way or another, so we know that it is the XML parser generating for us when the attribute is not specified (as opposed to that being just a magic value that we coordinate between XML and gamecode). Hopefully that makes sense.

Reply:
'NONE' is an explicitly defined parameter in the XML nothing to do with the XML parser or 'IMPLIED'. If the attribute is specified then you get an empty string. 'NONE' is used by map makers to specify the resource shouldn't be displayed for any players since the default (not specified, empty string) is that its displayed for all players.

Reply:
Ah, so it is a magic value then. Is there any difference between `NONE` and not specifying the attribute?

Reply:
Yeah, that's what I'm getting at. Not specifying (default) is you get an empty string which maps to showing the resource to all players. Specifying 'NONE' means the resource isn't displayed to any players. Reasoning being we want the default for all existing maps to be all players not none.

Reply:
Ok, makes sense now, thank you